### PR TITLE
Remove deprecated method which causing gcc10 warnings

### PR DESCRIPTION
**Pull request by linev**
Solving issue #253

### CODE REVIEW

File: Vc/scalar/vector.h [hunk 1]
```diff
@@ -210,13 +210,13 @@ template <typename T> class Vector<T, VectorAbi::Scalar>
...
-            typedef typename V2::EntryType AliasT2 Vc_MAY_ALIAS;
-            return V2(*reinterpret_cast<const AliasT2 *>(&m_data));
-        }
+//        template <typename V2>
+//        Vc_DEPRECATED("use reinterpret_components_cast instead") Vc_ALWAYS_INLINE V2
+//            reinterpretCast() const
+//        {
+//            typedef typename V2::EntryType AliasT2 Vc_MAY_ALIAS;
+//            return V2(*reinterpret_cast<const AliasT2 *>(&m_data));
+//        }
```
Comment (Axel-Naumann) [inline Vc/scalar/vector.h:219]: Why not just remove these lines?

Comment (linev) [reply Vc/scalar/vector.h:219]: To keep sign - something was there but just disabled

**Review by Axel-Naumann — APPROVED** [review]
In https://github.com/root-project/root/pull/5833 @amadio argued for removing only the deprecation message, while @linev suggests to remove the whole function signature. Given that the interface is deprecated [since 5 years](https://github.com/VcDevel/Vc/commit/663071a36ea87787257fa419a6bef069ee163aad) I think we should remove it. I bet Matthias deprecated it for a reason. I am OK with commenting it to leave a trace of the function's removal.

Given all this, I believe this PR should get merged.

**Comment by linev** [comment]
@Axel-Naumann, should we merge it? I cannot do it

**Pull request merged**